crates/blocks/src/registry.rs

```diff
@@ -35,7 +35,10 @@ impl BlockState {
     /// The returned `BlockData` is not linked with this `BlockState` instance.
     /// You need to call [`set_data`] to apply any changes made to the block data.
     pub fn data_as<T: BlockData>(self) -> Option<T> {
-        T::from_raw(&self.raw().properties)
+        T::from_raw(
+            &self.raw().properties,
+            self.get_valid_properties().to_owned(),
```

Comment:
We also want to avoid cloning the `ValidProperties` struct here, since it contains several allocations. Ideally, `&'static ValidProperties` would be more efficient.

Reply:
How could I achieve this? Should the `ValidProperties` structs get static lifetime when it's loaded while creating a new `BlockRegistry`?